nemo/collections/asr/data/audio_to_label.py

```diff
@@ -825,3 +997,761 @@ def sliced_seq_collate_fn(self, batch):
+    {"audio_filepath": "/path/to/audio_wav_n.wav", "num_speakers": 2,
+    "rttm_filepath": "/path/to/diar_label_n.rttm}
+
+    Args:
+        manifest_filepath (str):
+            Path to input manifest json files.
+        clus_label_dict (Dict):
+            Segment-level speaker labels from Clustering results.
+        soft_label_thres (float):
+            Threshold that determines the label of each segment based on RTTM file information.
+        featurizer:
+            Featurizer instance for generating features from the raw waveform.
+        window_stride (float):
+            Window stride for acoustic feature. This value is used for calculating the numbers of feature-level frames.
+        emb_batch_size (int):
+            Number of embedding vectors that are trained with attached computational graphs.
+        max_spks (int):
+            Integer value that limits the number of speakers for the model that is being trained.
+        bi_ch_infer (bool):
+            This variable should be True if dataloader is created for an inference task.
+    """
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """Returns definitions of module output ports.
+        """
+        output_types = {
+            "features": NeuralType(('B', 'T'), AudioSignal()),
+            "feature_length": NeuralType(('B'), LengthsType()),
+            "ms_seg_timestamps": NeuralType(('B', 'C', 'T', 'D'), LengthsType()),
+            "ms_seg_counts": NeuralType(('B', 'C'), LengthsType()),
+            "clus_label_index": NeuralType(('B', 'T'), LengthsType()),
+            "scale_mapping": NeuralType(('B', 'C', 'T'), LengthsType()),
+            "targets": NeuralType(('B', 'T', 'C'), ProbsType()),
+        }
+
+        return output_types
+
+    def __init__(
+        self,
+        *,
+        manifest_filepath: str,
+        multiscale_args_dict: str,
+        ms_ts_dict: Dict,
+        soft_label_thres: float,
+        featurizer,
+        window_stride,
+        emb_batch_size,
+        max_spks: int,
+        bi_ch_infer: bool,
+        random_flip: bool = True,
+    ):
+        super().__init__()
+        self.collection = DiarizationSpeechLabel(
+            manifests_files=manifest_filepath.split(','),
+            emb_dict=None,
+            clus_label_dict=None,
+            max_spks=max_spks,
+            bi_ch_infer=bi_ch_infer,
+        )
+        self.featurizer = featurizer
+        self.multiscale_args_dict = multiscale_args_dict
+        self.ms_ts_dict = ms_ts_dict
+        self.round_digits = 2
+        self.decim = 10 ** self.round_digits
+        self.fr_per_sec = 100
+        self.soft_label_thres = soft_label_thres
+        self.bi_ch_infer = bi_ch_infer
+        self.max_spks = 2
+        self.feat_per_sec = int(1 / window_stride)
+        self.emb_batch_size = emb_batch_size
+        self.random_flip = random_flip
+
+    def __len__(self):
+        return len(self.collection)
+
+    def s2n(self, x):
```

Comment:
can you move this diarization_utils or something or other utils file and use it here, so this function can be used elsewhere.

Also, name this to string_to_float

Reply:
move the whole processing to convert_rttm_line in diarization_utils.py

Reply:
Since we import model in `diarization_utils ` file, its getting into circular import error, can you try `speaker_utils.py`

Reply:
moved to speaker_utils.py